### PR TITLE
Ddce 1381

### PR DESCRIPTION
**Pull request by london-coder**
# DDCE-(StoryNumberHere)

##Bug fix / New feature (delete as appropriate)

Please include a summary / description of the change and which issue it fixes.  Include any relevant user needs, context or links to other PRs related to this PR (eg. acceptance tests, environment config).

## PR Suggestions
- Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?
- Have you done a visual check of the changes?
- Is there any still commented or unused code? Lingering printlns?
- Have things been implemented in a complicated way?
- Are the test still over the coverage threshold?
- Does the compiler throw a lot of warnings? 
- Have methods and tests been named clearly?
- Were there any changes in the config? Do changes need to be made in app-config-???

## Checklist PR Raiser
 - [ ]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [ ]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [ ]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [ ]  I've squashed my commits - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

## Checklist PR Reviewer
 - [ ]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [ ]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [ ]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [ ]  I've squashed my commits - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

File: project/plugins.sbt [hunk 1]
```diff
@@ -10,8 +10,6 @@ addSbtPlugin("uk.gov.hmrc" % "sbt-git-versioning" % "2.1.0")
 
 addSbtPlugin("uk.gov.hmrc" % "sbt-distributables" % "2.0.0")
 
-addSbtPlugin("uk.gov.hmrc" % "sbt-artifactory" % "1.5.0")
```
Comment (anburoach) [inline project/plugins.sbt:13]: Extracting structure fails without sbt-artifactory plugin... please add this

File: conf/app.routes [hunk 2]
```diff
@@ -46,3 +46,6 @@ GET       /we-cannot-find-your-application              controllers.CannotFindAp
 
 ### Switchover error page route
 GET       /error/problem-with-your-answers              controllers.SwitchOverErrorController.onPageLoad
+GET       /error/signed-you-out                         controllers.SignOutController.signedYouOut
```
Comment (anburoach) [inline conf/app.routes:49]: don't need 3 routes.. just one is enough... please keep the valid one

File: project/plugins.sbt (hunk 1 above)
Comment (london-coder) [reply project/plugins.sbt:13]: There was a message posted in Slack, that referenced the Confluence page:https://confluence.tools.tax.service.gov.uk/display/TEC/2021/01/18/Removing+the+sbt-artifactory+plugin

Comment (london-coder) [reply project/plugins.sbt:13]: I have no problems running the project locally with the plug in removed.

File: conf/application.conf [hunk 3]
```diff
@@ -137,6 +137,7 @@ urls {
   login = "http://localhost:9949/auth-login-stub/gg-sign-in"
   continue = "continue"
   signOut = "https://www.gov.uk/charity-recognition-hmrc"
+  signedYouOut = "https://www.gov.uk/charity-recognition-hmrc/error/signed-you-out"
```
Comment (anburoach) [inline conf/application.conf:140]: have to check with Blaine whether the url is correct or not

File: conf/messages.en [hunk 4]
```diff
@@ -73,7 +73,13 @@ error.submissionError.link = start your claim again
 session_expired.title = For your security, this service has been reset
 session_expired.heading = For your security, this service has been reset
 session_expired.guidance = The details you have given have been deleted because you did not continue the service for 15 minutes.
+session_expired.signedOut = For your security, we signed you out
```
Comment (anburoach) [inline conf/messages.en:76]: I don't see use of this message anywhere.. please remove it if it is not using

File: app/views/templates/GovukLayoutWrapper.scala.html [hunk 5]
```diff
@@ -58,7 +58,8 @@
                 timeout: @{appConfig.timeout},
                 countdown: @{appConfig.countdown},
                 keepAliveUrl: '@{controllers.routes.IndexController.keepalive().url}',
-                signOutUrl: '@{controllers.routes.SignOutController.signOut().url}'
+                signOutUrl: '@{controllers.routes.SignOutController.signOut().url}',
```
Comment (anburoach) [inline app/views/templates/GovukLayoutWrapper.scala.html:61]: plz modify the **signOutUrl** don't add new parameter **signedYouOutUrl**

File: app/controllers/SignOutController.scala [hunk 6]
```diff
@@ -19,12 +19,21 @@ package controllers
...
+import play.mvc.Http.Response
+import controllers.actions.SessionIdentifierAction
+import repositories.SessionRepositoryImpl
+import views.html.errors.SignedYouOutView
 
 
-class SignOutController @Inject()(val controllerComponents: MessagesControllerComponents,
-                                  appConfig: FrontendAppConfig) extends LocalBaseController {
+class SignOutController @Inject()(val controllerComponents: MessagesControllerComponents, identify: SessionIdentifierAction,
+                                  sessionRepository: SessionRepositoryImpl, view: SignedYouOutView)
```
Comment (anburoach) [inline app/controllers/SignOutController.scala:29]: remove unused declarations

File: app/controllers/SignOutController.scala [hunk 7]
```diff
@@ -19,12 +19,21 @@ package controllers
...
+                                  (implicit appConfig: FrontendAppConfig) extends LocalBaseController {
 
   def signOut: Action[AnyContent] = Action { _ =>
     Redirect(appConfig.signOutUrl).withNewSession
   }
+
+  def signedYouOut: Action[AnyContent] = Action { implicit request =>
```
Comment (anburoach) [inline app/controllers/SignOutController.scala:36]: Add unit test for this method

File: conf/application.conf (hunk 3 above)
Comment (london-coder) [reply conf/application.conf:140]: That is the URL specified in the ticket

File: conf/messages.en (hunk 4 above)
Comment (london-coder) [reply conf/messages.en:76]: Removed.

**Pull request closed without merging**